experimental/design-notes.txt

```diff
@@ -0,0 +1,112 @@
+that might need to set the value of potentially many LabelSets whereas in MM all Meters (in MM terminology) are bound to a single set of labels and will
+have exactly 1 data point per time period. Does the value of unbound instruments justify this additional complexity?
+
+5. MM has no batch mode for logging metrics, how valuable of a feature is this given its additional complexity to implement? Is it purely a performance optimization
+or is there a scenario where the atomicity of the measurements matter?
+
+6. MM has a lot of API surface dedicated to configuring Meters (in MM terminology) and tags. I am guessing this is the kind of functionality that would be sitting in OT's
+Views API?
+
+
+IfxMetricsExtensions:
+
+Is there a better doc than https://genevamondocs.azurewebsites.net/collect/instrument/ifx/ifxextensions.html? This page seems to assume I am already pretty familiar
+with Geneva metrics and only describes some specific addendums.
+
+Prometheus.NET - https://github.com/prometheus-net/prometheus-net
+
+1. Naming - What MM calls Meters and OT calls bound instruments, Prometheus calls Metrics.
+2. Similar to MM and different from OT, metrics can be created directly from static functions on a global Metrics class.
+3. Prometheus also has different metric types (instrument types in OT terminology). It has Counter, Guage, Summary, and Histogram. The best mapping appears to be:
+Counter       Counter
+Gauge         UpDownCounter or ValueRecorder?
+Summary       ValueRecorder
+Histogram     ValueRecorder
+NA            *Observer
+
+4. Prometheus uses Observe() API where OT uses Record() API. Prometheus doesn't appear to have asynchronous/callback style APIs.
+
+5. Prometheus has histogram and summaries configured with measurement precision and intervals at the point they are defined. OT and MM configure this via a
+separate configuration API.
+
+6. Prometheus has various helper utilities like NewTimer() that that handles details of timing code and then inputs the timed value into Histogram. This
+appears to avoid needing to declare a dedicated Timer Meter as MM did.
+
+
+
+AppMetrics - https://www.app-metrics.io/
+
+1. Naming - What OT calls instruments, AppMetrics calls metrics
+2. AppMetrics has an IMetrics interface somewhat similar to MeterProvider or MetricsRegistry in other libraries, however the provider isn't responsible for directly
+creating the API objects that represent metrics. Instead Metric definitions are created independent from any builder (similar to a LabelSet) and then passed as an
+argument to a measurement recording function. For example:
+IMetrics metrics = ...
+var counter = new CounterOptions { Name = "my_counter" } // this defines the metric
+metrics.Measure.Counter.Increment(counter); // increments the value of the metric stored in this IMetrics
+
+3. Metrics are never bound. Tags can be provided at the time a measurement is captured and merged with tags set via global tags in the IMetrics interface
```

Comment:
I'm wondering... Maybe we setup a memory storage for binding a set of KVs.  Then, we just pass the id of that entry, instead of "binding" a meter.  This way, any "bound" KVs can be shared by any meter and addressible with an id.